In [1]:
import numpy as np
import pandas as pd

In [2]:
from sqlalchemy import create_engine
from sqlalchemy import text
postgres_str = 'postgresql://postgres:docker@localhost:5432/enceladus'
connect = create_engine(postgres_str)

In [3]:
#Master Plan Data
pd.read_sql_query ("""select * from events
master_plan;""", connect).head(10)

id          time_stamp                                        title  \
0   1 2004-10-07 20:30:00                     Phoebe Spectrophotometry   
1   2 2008-06-25 07:50:00     FP3 coverage of Dione zero phase passage   
2   3 2008-12-17 09:30:00                  CIRS_097RH_020W151PH001_ISS   
3   4 2008-12-17 09:30:00                               spectral cubes   
4   5 2008-12-17 09:30:00                          Rhea phase coverage   
5   6 2007-10-26 12:25:00  Enceladus spectrophotometry/ phase coverage   
6   7 2007-10-26 12:25:00                                         None   
7   8 2007-10-26 11:30:00       Rhea spectrophotometry/ phase coverage   
8   9 2007-10-26 11:30:00                                         None   
9  10 2007-10-26 17:40:00      Mimas spectrophotometry/ phase coverage   

                                         description  event_type_id  \
0  Turn -Y to Phoebe, do during one Phoebe rotati...          482.0   
1  FP3 ISS rider on Dione zero phase sequence.  A...          544.0   
2                                              rider            NaN   
3                                        Phase study            NaN   
4  Turn UVIS_FUV to Rhea; do ftprt with (0,0,-1.0...            NaN   
5    Turn -Y to Enceladus; do 1x1 mosaic; turn to WP          405.0   
6                                               None          405.0   
7  Turn -Y to Rhea; do 1x2 mosaic (2nd ftprt offs...          405.0   
8                                               None          405.0   
9        Turn -Y to Mimas; do 1x1 mosaic; turn to WP          405.0   

   target_id  team_id  request_id  spass_type_id  
0         45       13        2636            2.0  
1         20       10        4962            5.0  
2          1       10         991            5.0  
3          1        8         991            5.0  
4          1       13         991            2.0  
5         28       13        3958            2.0  
6         28        4        3958            5.0  
7          1       13        2296            2.0  
8          1        4        2296            5.0  
9         22       13        2296            2.0

Flybys

In [4]:
statement = """select targets.description as target, time_stamp,
title
from events
inner join targets on target_id=targets.id where title ilike '%flyby%'
or title ilike '%fly by%';"""
pd.read_sql_query(text(statement), connect).head(20)


target          time_stamp  \
0             Aegaeon 2015-12-19 12:00:00   
1               Atlas 2017-04-12 11:30:00   
2               Atlas 2015-12-06 21:08:00   
3             Iapetus 2007-09-10 12:30:40   
4               Mimas 2005-08-02 01:20:02   
5              Tethys 2007-08-29 14:40:00   
6             Iapetus 2004-12-31 19:00:00   
7             Iapetus 2004-12-31 22:30:00   
8               Dione 2005-10-11 18:07:22   
9               Dione 2011-12-11 21:39:22   
10              Dione 2011-12-12 11:39:22   
11              Dione 2015-06-16 08:11:52   
12              Dione 2015-06-16 22:30:00   
13              Dione 2015-08-17 06:33:26   
14              Dione 2015-08-17 20:33:26   
15              Dione 2010-04-07 04:53:11   
16  DustRAM direction 2015-06-16 17:30:00   
17              Dione 2005-10-11 17:44:02   
18              Dione 2009-07-11 01:12:32   
19              Dione 2009-09-20 14:52:04   

                                                title  
0                                       Aegaeon flyby  
1                             14,000km flyby of Atlas  
2                      Flybys of Atlas and Prometheus  
3                  Iapetus I49 targeted flyby imaging  
4                               MIMAS FLYBY(NT) Color  
5                                  TETHYS FLYBY Color  
6     Iapetus close flyby 00B/00C local-color imaging  
7         Iapetus close flyby 00B/00C context imaging  
8   Dione targeted flyby high-phase high-res and S...  
9                                         Dione flyby  
10                                        Dione flyby  
11                                        Dione flyby  
12                                        Dione flyby  
13                                        Dione flyby  
14                                        Dione flyby  
15                                        Dione Flyby  
16          CDA D4 Flyby with E-Ring background wings  
17             Dione targeted flyby C/A rider imaging  
18                                           D2 flyby  
19                                           D3 flyby

We have a titan flyby for feb 15th see above, the second Enceladus flyby on March 9.. where is the first one ?

In [5]:
pd.read_sql_query ("""select target, title, date
from import.master_plan
where start_time_utc::date = '2005-02-17' order by start_time_utc::date;""", connect).head(20)
# this data was imported into a table called master plan which we can also check with sql

target                                              title       date
0   Enceladus                                          Enceladus  17-Feb-05
1   Enceladus  Enceladus FP1 and FP3 maps, high spectral reso...  17-Feb-05
2   Enceladus  ICYLON:  Icy Satellite Longitude / Phase Coverage  17-Feb-05
3      Saturn                                        MAPS Survey  17-Feb-05
4      Saturn    Obtain wideband examples of lightning whistlers  17-Feb-05
5      Saturn                Highest Resolution SED Observations  17-Feb-05
6   Enceladus                                          Enceladus  17-Feb-05
7   Enceladus  Enceladus rider (FP1,FP3 high spectral resolut...  17-Feb-05
8   Enceladus  ICYLON:  Icy Satellite Longitude / Phase Coverage  17-Feb-05
9   Enceladus                          ENCELADUS Limb Topography  17-Feb-05
10  Enceladus                 Enceladus rider (FP1,FP3 coverage)  17-Feb-05
11  Enceladus         ICYMAP:  High resolution icy satellite map  17-Feb-05
12  Enceladus                                          Enceladus  17-Feb-05
13  Enceladus                                MAPS 003EN Campaign  17-Feb-05
14  Enceladus        RSS Enceladus Gravity Thermal Stabilization  17-Feb-05
15  Enceladus            Enceladus closest approach observations  17-Feb-05
16  Enceladus                     Dust envelope around Enceladus  17-Feb-05
17  Enceladus                     ENCELADUS near-limb topography  17-Feb-05
18  Enceladus                  ICYEXO:  Icy Satellite Exospheres  17-Feb-05
19      Other                       MAG_MAPS_Survey_Measurements  17-Feb-05

If you see above the record 4 says "Obtain wideband examples of lightning whistlers" What the heck is a lightning whistler? Unfortunately, there’s nothing in the data that suggests this is a flyby

"This map of the surface of Saturn's moon Enceladus illustrates the regions that will be imaged by Cassini during the spacecraft's first very close flyby of the moon on Feb. 17, 2005. At closest approach, the spacecraft is expected to pass approximately 1,180 kilometers (733 miles) above the moon's surface. Enceladus is 505 kilometers (314 miles) across."
It was never actually given a name, however, and the flyby that occurred just after this one in March of 2005 was named E-1. I guess the February 2005 one is Rob’s zeroth flyby. They could’ve made that a little easier.

This is when you evaluate data like this and see if you need it or try to find out what happened with it and what actions can you take.


Data investigation is fun, isn’t it?


![](Images/firstat.png)

In [6]:
pd.read_sql_query('''select
targets.description as target, events.time_stamp, event_types.description as event
from events
inner join event_types on event_types.id = events.event_type_id
inner join targets on targets.id = events.target_id
where events.time_stamp::date='2005-02-17'
and targets.id = 28
order by events.time_stamp;''', connect).head(20)

target          time_stamp                                   event
0   Enceladus 2005-02-17 00:00:29                               Enceladus
1   Enceladus 2005-02-17 00:15:29     Icy satellite longitudinal coverage
2   Enceladus 2005-02-17 00:15:29          CIRS FP1 integration / FP3 map
3   Enceladus 2005-02-17 01:30:29                               Enceladus
4   Enceladus 2005-02-17 01:30:29                               Enceladus
5   Enceladus 2005-02-17 01:30:29     Icy satellite longitudinal coverage
6   Enceladus 2005-02-17 02:15:29               Icy satellite surface map
7   Enceladus 2005-02-17 02:15:29                               Enceladus
8   Enceladus 2005-02-17 02:15:29                         Limb topography
9   Enceladus 2005-02-17 02:15:29                         Limb topography
10  Enceladus 2005-02-17 03:00:29  Enceladus closest approach observation
11  Enceladus 2005-02-17 03:00:29                           MAPS Campaign
12  Enceladus 2005-02-17 03:00:29            Thermal stabilization period
13  Enceladus 2005-02-17 03:05:29                 Icy satellite exosphere
14  Enceladus 2005-02-17 03:05:29              Enceladus dust observation
15  Enceladus 2005-02-17 03:05:29              Enceladus dust observation
16  Enceladus 2005-02-17 04:00:29     Icy satellite longitudinal coverage
17  Enceladus 2005-02-17 04:00:29                               Enceladus
18  Enceladus 2005-02-17 04:00:29                  CIRS FP1 dark side map
19  Enceladus 2005-02-17 05:25:29         RSS gravity field determination

In [7]:
pd.read_sql_query('''select
id, time_stamp, title, description from enceladus_events where date ='2005-02-17';''', connect)

id          time_stamp  \
0   14654 2005-02-17 00:00:29   
1   22660 2005-02-17 00:15:29   
2   18689 2005-02-17 00:15:29   
3   14650 2005-02-17 01:30:29   
4   22658 2005-02-17 01:30:29   
5   14646 2005-02-17 01:30:29   
6   30305 2005-02-17 02:15:29   
7   14659 2005-02-17 02:15:29   
8   23136 2005-02-17 02:15:29   
9   30306 2005-02-17 02:15:29   
10  14778 2005-02-17 03:00:29   
11  14409 2005-02-17 03:00:29   
12  58854 2005-02-17 03:00:29   
13  14426 2005-02-17 03:05:29   
14  21924 2005-02-17 03:05:29   
15  14425 2005-02-17 03:05:29   
16  14656 2005-02-17 04:00:29   
17  18605 2005-02-17 04:00:29   
18  22657 2005-02-17 04:00:29   
19  20164 2005-02-17 05:25:29   
20  47247 2005-02-17 08:30:29   
21  41501 2005-02-17 11:30:29   
22  41500 2005-02-17 11:30:29   
23  22654 2005-02-17 11:30:29   

                                                title  \
0                                           Enceladus   
1   ICYLON:  Icy Satellite Longitude / Phase Coverage   
2   Enceladus FP1 and FP3 maps, high spectral reso...   
3   Enceladus rider (FP1,FP3 high spectral resolut...   
4   ICYLON:  Icy Satellite Longitude / Phase Coverage   
5                                           Enceladus   
6                  Enceladus rider (FP1,FP3 coverage)   
7                                           Enceladus   
8          ICYMAP:  High resolution icy satellite map   
9                           ENCELADUS Limb Topography   
10                                MAPS 003EN Campaign   
11            Enceladus closest approach observations   
12        RSS Enceladus Gravity Thermal Stabilization   
13                     ENCELADUS near-limb topography   
14                  ICYEXO:  Icy Satellite Exospheres   
15                     Dust envelope around Enceladus   
16                                          Enceladus   
17                         FP1 map of dark hemisphere   
18  ICYLON:  Icy Satellite Longitude / Phase Coverage   
19          RSS Enceladus Gravity Field Determination   
20          Scatterometry and Radiometry of Enceladus   
21    Enceladus rider (FP3 0.5 cm-1 res. integration)   
22                            ENCELADUS Geyser Search   
23  ICYLON:  Icy Satellite Longitude / Phase Coverage   

                                          description  
0   64i2-64v1 160ms, 640ms (28 min/cycle) 21 Mbits...  
1   Ride along with other ORS instrument icy satel...  
2   NOTE: SHORTENED BY 15 MIN., SO DESIGN MUST BE ...  
3   FP1, FP3 observation of Enceladus to search fo...  
4   Ride along with other ORS instrument icy satel...  
5   64x64i2-v3 160ms, 320ms, 80ms (28 min/cycle) 2...  
6   ISS rider.  FP1, FP3 observation of Enceladus ...  
7   24x12i2-v1 160ms, 1 min/cycle) 0.73 Mbits/fram...  
8                   Ride along on high resolution map  
9                                   5 NAC GRN filters  
10  Part of 003EN MAPS Campaign.  Observe interact...  
11  RPWS observations in the immediate vicinity of...  
12  A period of thermal stabilization is required ...  
13  9 NAC GRN filters. Will attempt to snap a seri...  
14   Target to lambda Sco as part of CDA observation.  
15  study E ring dust distribution, poiting:  prim...  
16  64i1-64v1 160, 320ms, 34 min/cycle) 21 Mbits/s...  
17  FP1 dark side map of Enceladus. Phase= 156, Lo...  
18  Ride along with other ORS instrument icy satel...  
19                       Enceladus mass determination  
20  Point -Z axes at target and execute raster sca...  
21  ISS rider in staring mode.  FP3,4 observation ...  
22                            Filterset = wplume (5);  
23  Ride along with other ORS instrument icy satel...

![](Images/enceladus.png)

Enceladus rider (FP1,FP3 high spectral resolution integration) the plan regard- ing “FP3 observation”. That’s from CIRS, the Composite Infrared Spectrometer. FP3 is the mid-range focal plane which is used to detect volcanism.

This one
FP1 dark side map of Enceladus. Phase= 156, Local time= 0100, sub S/C=(2, 162), AD=23 mrad. Turn from waypoint (15 min?); 14 scan Nyqiust FP1 map of disk at 15.5 cm-1 resolu- tion (4.75 sec/frame) (25 min); return to waypoint? (14 min?) (they were tryng to align cassini its imaging systems where Enceladus is positioned between Cassini and a star somewhere. They were looking for the plumes, hoping distant starlight would light up the icy spray

Cassini’s magnetometer (MAG), however, did capture some- thing surprising:
“It was like the magnetic field lines of Saturn, which were moving towards Enceladus, were being draped upstream of Enceladus, al- most like there was an obstacle [above the moon] ... but a larger obstacle than the body itself"

Michele Dougherty. put her reputation on the line to discover more about Ence- ladus, what a star! If she hadn’t done that, we might have missed out entirely on this little moon

![](Images/enceladuso.png)

Enceladus became something more than a “dead ice ball.”

Every flyby tells a story about Enceladus, and since Enceladus is what SELFI is all about, we need to know, to a very full extent, what happened when, with respect to that curious moon. F

That flyby happened on February 17, 2005, and the results did not make sense. Every other instrument showed nothing, except for the magnetometer. The mystery deepened and another flyby was scheduled weeks later on March 9, 2005.

In [8]:
#after there was the Gold trade mission plan where they took another look of enceladus with every instrument to verify 
# what they saw -- 14410	2005-03-09 08:38:01	Enceladus closest approach observationsb
pd.read_sql_query('''select id, time_stamp, title,description
from enceladus_events where date ='2005-03-09';''',connect)

id          time_stamp  \
0   14614 2005-03-09 00:00:00   
1   18875 2005-03-09 01:48:01   
2   14643 2005-03-09 01:48:01   
3   23156 2005-03-09 01:48:01   
4   23142 2005-03-09 02:52:01   
5   14632 2005-03-09 02:52:01   
6   37630 2005-03-09 02:52:01   
7   44693 2005-03-09 02:52:01   
8   14629 2005-03-09 03:07:01   
9   18869 2005-03-09 03:07:01   
10  23152 2005-03-09 03:07:01   
11  37635 2005-03-09 04:07:01   
12  23150 2005-03-09 04:07:01   
13  14625 2005-03-09 04:07:01   
14  18594 2005-03-09 04:37:01   
15  14624 2005-03-09 04:37:01   
16  23188 2005-03-09 04:37:01   
17  37639 2005-03-09 05:27:01   
18  14621 2005-03-09 05:27:01   
19  23165 2005-03-09 05:27:01   
20  18868 2005-03-09 05:52:01   
21  14612 2005-03-09 05:52:01   
22  23161 2005-03-09 05:52:01   
23  23134 2005-03-09 06:47:01   
24  14617 2005-03-09 06:47:01   
25  14957 2005-03-09 07:08:01   
26  14666 2005-03-09 07:27:01   
27  18595 2005-03-09 07:27:01   
28  23159 2005-03-09 07:27:01   
29  14398 2005-03-09 07:38:01   
30  43570 2005-03-09 07:47:01   
31  23185 2005-03-09 07:47:01   
32  14663 2005-03-09 07:47:01   
33  14958 2005-03-09 08:05:00   
34  23417 2005-03-09 08:38:01   
35  14770 2005-03-09 08:38:01   
36  14410 2005-03-09 08:38:01   
37  14765 2005-03-09 08:39:00   
38  23414 2005-03-09 08:53:01   
39  14661 2005-03-09 08:58:01   
40    213 2005-03-09 09:10:05   
41  23418 2005-03-09 09:23:01   
42  14651 2005-03-09 10:38:01   
43  18710 2005-03-09 10:38:01   
44  18709 2005-03-09 10:42:01   
45  23121 2005-03-09 10:42:01   
46  47252 2005-03-09 12:08:01   
47    221 2005-03-09 13:55:12   
48  14644 2005-03-09 18:10:00   
49  22591 2005-03-09 18:25:00   

                                                title  \
0                                Enceladus CIRS Rider   
1                         Enceladus 004 FP3 map No. 1   
2                                  Enceladus targeted   
3     ICYMAP:  UV Surface Map of Enceladus, ORS Rider   
4     ICYMAP:  UV Surface Map of Enceladus, ORS Rider   
5                                  Enceladus targeted   
6                            NAC 3 COLOR POLARIZATION   
7                                                 TBD   
8                                  Enceladus targeted   
9                             Enceladus FP3 map No. 2   
10    ICYMAP:  UV Surface Map of Enceladus, ORS Rider   
11                                         NAC 4-COLR   
12    ICYMAP:  UV Surface Map of Enceladus, ORS Rider   
13                                 Enceladus targeted   
14  FP1 disk integration at high spectral resolution.   
15                                 Enceladus targeted   
16    ICYMAP:  UV Surface Map of Enceladus, ORS Rider   
17                                        NAC 4 COLOR   
18                                 Enceladus targeted   
19    ICYMAP:  UV Surface Map of Enceladus, ORS Rider   
20                        Enceladus 004 FP3 map No. 3   
21                                 Enceladus targeted   
22    ICYMAP:  UV Surface Map of Enceladus, ORS Rider   
23               ICYMAP:  UV Surface Map of Enceladus   
24                                 Enceladus targeted   
25                           Enceladus targeted flyby   
26                                 Enceladus targeted   
27                     Enceladus 004 FP1 dayside map.   
28    ICYMAP:  UV Surface Map of Enceladus, ORS Rider   
29                     Dust envelope around Enceladus   
30                 NAC 3x3x1 GRN MOSAIC + WAC 4 Color   
31               ICYMAP:  UV Surface Map of Enceladus   
32                                 Enceladus targeted   
33                           Enceladus targeted flyby   
34                                               SOST   
35                                   MAPS RS Campaign   
36            Enceladus closest approach observations   
37                                MAPS 004EN Campaign   
38                                               SOST   
39                                      RSS

The RPWS was the Radio and Plasma Wave Science instrument that, duh, was all about radio and plasma waves. It also studied dust particles, but that wasn’t important enough to go in the name apparently. The team was very curious about this “atmosphere” that the MAG reported, and they needed more data, so they started poking around for bits of anything floating in space.

In [9]:
# Investigations were happening regarding thermal activity so we will use a full text queries
enceladus_event='''select id, date, title from enceladus_events where date
between '2005-02-01'::date and '2005-02-28'::date
and search @@ to_tsquery('thermal');'''
pd.read_sql_query(text(enceladus_event), connect)

id        date                                              title
0  19345  2005-02-16                  Enceladus rider (FP3 integration)
1  58853  2005-02-16        RSS Enceladus Gravity Thermal Stabilization
2  19344  2005-02-16                  Enceladus rider (FP3 integration)
3  18689  2005-02-17  Enceladus FP1 and FP3 maps, high spectral reso...
4  14650  2005-02-17  Enceladus rider (FP1,FP3 high spectral resolut...
5  30305  2005-02-17                 Enceladus rider (FP1,FP3 coverage)
6  58854  2005-02-17        RSS Enceladus Gravity Thermal Stabilization
7  41501  2005-02-17    Enceladus rider (FP3 0.5 cm-1 res. integration)

In [10]:
enceladus_event='''select id, date, title from enceladus_events where date
between '2005-03-01'::date and '2005-03-30'::date
and search @@ to_tsquery('thermal');'''
pd.read_sql_query(text(enceladus_event), connect)

#there was not anyt thermal activities in march

Empty DataFrame
Columns: [id, date, title]
Index: []

Each instrument aboard Cassini had a dedicated team back on earth. The mission plan details which team is, basically, “driving Cassini” at any given time.

In [11]:
pd.read_sql_query(''' select title, teams.description
from events
inner join teams on teams.id=team_id where time_stamp::date='2005-03-09' and target_id=28;''',connect).head(20)

title description
0   Enceladus Orbit Xing 0.00<lat< 0.50 deg 3.85.....         CDA
1   Enceladus Orbit Xing 0.00<lat< 0.50 deg 3.85.....         CDA
2                      Dust envelope around Enceladus         CDA
3             Enceladus closest approach observations        RPWS
4                                  Enceladus targeted        VIMS
5                                Enceladus CIRS Rider        CIRS
6                                  Enceladus targeted        VIMS
7                                  Enceladus targeted        VIMS
8                                  Enceladus targeted        VIMS
9                                  Enceladus targeted        VIMS
10                                 Enceladus targeted        VIMS
11                                 Enceladus targeted        VIMS
12                                 Enceladus targeted        VIMS
13                                 Enceladus targeted        VIMS
14                                 Enceladus targeted        VIMS
15                                      RSS_Ridealong        CIRS
16                                 Enceladus targeted        VIMS
17                                 Enceladus targeted        VIMS
18                                MAPS 004EN Campaign        CAPS
19                                   MAPS RS Campaign        MIMI

In [12]:
#looks like all were ative at some point lets check who had the most
pd.read_sql_query('''
select count(1) as activity, teams.description from events
inner join teams on teams.id=team_id
where time_stamp::date='2005-03-09'
and target_id=28
group by teams.description
order by activity desc; ''',connect)



activity description
0         14        CIRS
1         12        UVIS
2         12        VIMS
3          7         ISS
4          3         CDA
5          3        INMS
6          2         MAG
7          1       RADAR
8          1        RPWS
9          1        MIMI
10         1        CAPS

During the second flyby, on March 9, 2005, the Com- posite Infrared Scanner (CIRS) was the most active. So much for thermal, huh? The team was looking for a source of heat, some- thing that would cause the eruption of ice and water into space.

In [13]:
#materialized view that has an index in it for a better way to search, only way to do a view with an index is by 
# using a materlialized view (they are great for speeding things up)

pd.read_sql_query(''' select id, date,title
from enceladus_events
where search @@ to_tsquery('closest');''', connect)

id        date                                            title
0   14409  2005-02-17          Enceladus closest approach observations
1   14410  2005-03-09          Enceladus closest approach observations
2   14411  2005-07-14          Enceladus closest approach observations
3   14424  2008-03-12          Enceladus closest approach observations
4    9777  2008-03-12  Drag CIRS FOVs across disk at closest approach.
5   14405  2008-08-11          Enceladus closest approach observations
6   14401  2008-10-09          Enceladus closest approach observations
7   14406  2008-10-31          Enceladus closest approach observations
8   14407  2009-11-02          Enceladus closest approach observations
9   14404  2009-11-02          Enceladus closest approach observations
10  14403  2009-11-21          Enceladus closest approach observations
11  14408  2010-04-27          Enceladus closest approach observations
12  14402  2010-05-18          Enceladus closest approach observations
13  14416  2010-08-13          Enceladus closest approach observations
14  14417  2010-11-30          Enceladus closest approach observations
15  14418  2010-12-21          Enceladus closest approach observations
16  14412  2011-10-01          Enceladus closest approach observations
17  14419  2011-10-19          Enceladus closest approach observations
18  14414  2011-11-06          Enceladus closest approach observations
19  14413  2012-03-27          Enceladus closest approach observations
20  14415  2012-04-14          Enceladus closest approach observations
21  14420  2012-05-02          Enceladus closest approach observations
22  14422  2015-10-14          Enceladus closest approach observations
23  14421  2015-10-28          Enceladus closest approach observations
24  14423  2015-12-19          Enceladus closest approach observations

In [14]:
# closes approach two times in november.. Let's dig into
pd.read_sql_query ('''select time_stamp, title
from events
where time_stamp::date='2009-11-02' order by time_stamp;''', connect)


time_stamp                                              title
0  2009-11-02 00:44:00                    MAG_Internal_Field_Measurements
1  2009-11-02 00:44:00                               MAG MAPS RS Campaign
2  2009-11-02 00:50:00                                          Data rate
3  2009-11-02 01:13:58              Scatterometry/Radiometry of Enceladus
4  2009-11-02 04:38:58                                               None
5  2009-11-02 05:42:00                                Enceladus encounter
6  2009-11-02 06:41:58                                MAPS 120EN Campaign
7  2009-11-02 06:42:58                                Enceladus encounter
8  2009-11-02 06:42:58                        Enceladus Flyby C/A imaging
9  2009-11-02 06:44:00                                          Data rate
10 2009-11-02 07:00:00                     ENC INMS-CDA 2009-306T0:42 MMH
11 2009-11-02 07:11:58            Enceladus closest approach observations
12 2009-11-02 07:11:58                                   MAPS RS Campaign
13 2009-11-02 07:36:58                         CIRS_120EN_ENCEL7001_INMS;
14 2009-11-02 07:39:58                            ICYMAP:  UV Surface Map
15 2009-11-02 07:56:58            Enceladus closest approach observations
16 2009-11-02 08:03:58                            ICYMAP:  UV Surface Map
17 2009-11-02 08:03:58                                Enceladus encounter
18 2009-11-02 08:03:58                                          Enceladus
19 2009-11-02 08:03:58                      CIRS_120EN_FP3SPMAP001_PRIME;
20 2009-11-02 08:03:58                                          Enceladus
21 2009-11-02 08:11:58                 MIMI Satellite & Ring Interactions
22 2009-11-02 08:11:58                           RPWS Ring Plane Crossing
23 2009-11-02 08:13:58              ICYMAP:  UV of plume occulting Saturn
24 2009-11-02 08:13:58                          CIRS_120EN_ENUVIS001_UVIS
25 2009-11-02 08:13:58                                          Enceladus
26 2009-11-02 08:34:00                           Enceladus targeted flyby
27 2009-11-02 08:41:58                                        MAPS Survey
28 2009-11-02 09:35:00                               RPWS CHORUS Campaign
29 2009-11-02 09:41:58                        CIRS_120EN_FP1STARE001_ENGR
30 2009-11-02 09:41:58              ICYMAP:  UV of Plume Occulting Saturn
31 2009-11-02 09:41:58                                          Enceladus
32 2009-11-02 09:42:00                            Segment Boundary Survey
33 2009-11-02 09:42:00                               MAG MAPS RS Campaign
34 2009-11-02 10:03:58                                          Enceladus
35 2009-11-02 10:03:58                      CIRS_120EN_ENFP1MAP001_PRIME;
36 2009-11-02 10:03:58  ICYLON:  Icy Satellite Longitude / Phase Coverage
37 2009-11-02 11:44:00  ICYLON:  Icy Satellite Longitude / Phase Coverage
38 2009-11-02 11:44:00                          CIRS_120EN_ENVIMS001_VIMS
39 2009-11-02 11:44:00                                          Enceladus
40 2009-11-02 14:04:00  ICYLON:  Icy Satellite Longitude / Phase Coverage
41 2009-11-02 14:04:00                          CIRS_120TE_ICYLON001_UVIS
42 2009-11-02 14:04:00                                             Tethys
43 2009-11-02 14:04:00                                             Tethys
44 2009-11-02 16:00:00                                  RPWS Inner Survey
45 2009-11-02 17:00:00                                 Data rate End:SApo
46 2009-11-02 17:00:00                       GSE pass for Enceladus' mass
47 2009-11-02 17:00:00                UVIS Interplanetary Hydrogen Survey
48 2009-11-02 18:00:00                           CIRS_120IC_DSCAL09307_SP

In [15]:
# two flights by 45 minutes difference ? ummm  --Enceladus closest approach observations

# 2009-11-02 08:34:00	 Enceladus Targeted flyby - everydate gas -08 maybe an UTC time?



In [17]:
pd.read_sql_query ('''select (time_stamp at time zone 'UTC'), title
from events
where (time_stamp at time zone 'UTC')::date='2009-11-02'
order by time_stamp;''', connect)
# same results


timezone  \
0  2009-11-02 00:44:00+00:00   
1  2009-11-02 00:44:00+00:00   
2  2009-11-02 00:50:00+00:00   
3  2009-11-02 01:13:58+00:00   
4  2009-11-02 04:38:58+00:00   
5  2009-11-02 05:42:00+00:00   
6  2009-11-02 06:41:58+00:00   
7  2009-11-02 06:42:58+00:00   
8  2009-11-02 06:42:58+00:00   
9  2009-11-02 06:44:00+00:00   
10 2009-11-02 07:00:00+00:00   
11 2009-11-02 07:11:58+00:00   
12 2009-11-02 07:11:58+00:00   
13 2009-11-02 07:36:58+00:00   
14 2009-11-02 07:39:58+00:00   
15 2009-11-02 07:56:58+00:00   
16 2009-11-02 08:03:58+00:00   
17 2009-11-02 08:03:58+00:00   
18 2009-11-02 08:03:58+00:00   
19 2009-11-02 08:03:58+00:00   
20 2009-11-02 08:03:58+00:00   
21 2009-11-02 08:11:58+00:00   
22 2009-11-02 08:11:58+00:00   
23 2009-11-02 08:13:58+00:00   
24 2009-11-02 08:13:58+00:00   
25 2009-11-02 08:13:58+00:00   
26 2009-11-02 08:34:00+00:00   
27 2009-11-02 08:41:58+00:00   
28 2009-11-02 09:35:00+00:00   
29 2009-11-02 09:41:58+00:00   
30 2009-11-02 09:41:58+00:00   
31 2009-11-02 09:41:58+00:00   
32 2009-11-02 09:42:00+00:00   
33 2009-11-02 09:42:00+00:00   
34 2009-11-02 10:03:58+00:00   
35 2009-11-02 10:03:58+00:00   
36 2009-11-02 10:03:58+00:00   
37 2009-11-02 11:44:00+00:00   
38 2009-11-02 11:44:00+00:00   
39 2009-11-02 11:44:00+00:00   
40 2009-11-02 14:04:00+00:00   
41 2009-11-02 14:04:00+00:00   
42 2009-11-02 14:04:00+00:00   
43 2009-11-02 14:04:00+00:00   
44 2009-11-02 16:00:00+00:00   
45 2009-11-02 17:00:00+00:00   
46 2009-11-02 17:00:00+00:00   
47 2009-11-02 17:00:00+00:00   
48 2009-11-02 18:00:00+00:00   

                                                title  
0                     MAG_Internal_Field_Measurements  
1                                MAG MAPS RS Campaign  
2                                           Data rate  
3               Scatterometry/Radiometry of Enceladus  
4                                                None  
5                                 Enceladus encounter  
6                                 MAPS 120EN Campaign  
7                                 Enceladus encounter  
8                         Enceladus Flyby C/A imaging  
9                                           Data rate  
10                     ENC INMS-CDA 2009-306T0:42 MMH  
11            Enceladus closest approach observations  
12                                   MAPS RS Campaign  
13                         CIRS_120EN_ENCEL7001_INMS;  
14                            ICYMAP:  UV Surface Map  
15            Enceladus closest approach observations  
16                            ICYMAP:  UV Surface Map  
17                                Enceladus encounter  
18                                          Enceladus  
19                      CIRS_120EN_FP3SPMAP001_PRIME;  
20                                          Enceladus  
21                 MIMI Satellite & Ring Interactions  
22                           RPWS Ring Plane Crossing  
23              ICYMAP:  UV of plume occulting Saturn  
24                          CIRS_120EN_ENUVIS001_UVIS  
25                                          Enceladus  
26                           Enceladus targeted flyby  
27                                        MAPS Survey  
28                               RPWS CHORUS Campaign  
29                        CIRS_120EN_FP1STARE001_ENGR  
30              ICYMAP:  UV of Plume Occulting Saturn  
31                                          Enceladus  
32                            Segment Boundary Survey  
33                               MAG MAPS RS Campaign  
34                                          Enceladus  
35                      CIRS_120EN_ENFP1MAP001_PRIME;  
36  ICYLON:  Icy Satellite Longitude / Phase Coverage  
37  ICYLON:  Icy Satellite Longitude / Phase Coverage  
38                          CIRS_120EN_ENVIMS001_VIMS  
39                                          Enceladus  
40  ICYLON:  Icy Satellite Longitude / Phase Coverage  
41                          CIRS_120TE_ICYLON001_UVIS  
42         

In [19]:
pd.read_sql_query(''' select 19000/(pi() * 313) as orbits_per_hour''', connect)

orbits_per_hour
0        19.322325

To Be continue!